tiledb/sm/query/readers/aggregators/count_aggregator.cc

```diff
@@ -96,4 +88,18 @@ void CountAggregator::copy_to_user_buffer(
   }
 }
 
+NullCountAggregator::NullCountAggregator(FieldInfo field_info)
+    : CountAggregatorBase(field_info)
```

Comment:
The architecture here is slightly off, `NullCountAggregator` should be the derived that has a constructor based on `FieldInfo` (which also performs the checks) and `CountAggregatorBase`, `CountAggregator` should have a default constructor. This will avoid the weird `FieldInfo("","",false)` which tries to replace a default constructed FieldInfo which we explicitly deleted.